x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings_management/index.tsx

```diff
@@ -0,0 +1,262 @@
+    onSelectedConversationChange,
+  });
+
+  const onEditActionClicked = useCallback(
+    (rowItem: ConversationTableItem) => {
+      openEditFlyout();
+      onConversationSelectionChange(rowItem);
+    },
+    [onConversationSelectionChange, openEditFlyout]
+  );
+
+  const onConversationDeleted = useConversationDeleted({
+    conversationSettings,
+    conversationsSettingsBulkActions,
+    setConversationSettings,
+    setConversationsSettingsBulkActions,
+  });
+
+  const onDeleteActionClicked = useCallback(
+    (rowItem: ConversationTableItem) => {
+      setDeletedConversation(rowItem);
+      onConversationDeleted(rowItem.title);
+
+      closeEditFlyout();
+      openConfirmModal();
+    },
+    [closeEditFlyout, onConversationDeleted, openConfirmModal]
+  );
+
+  const onDeleteConfirmed = useCallback(() => {
+    if (Object.keys(conversationsSettingsBulkActions).length === 0) {
+      return;
+    }
+    closeConfirmModal();
+    handleSave(true);
+    setConversationsSettingsBulkActions({});
+  }, [
+    closeConfirmModal,
+    conversationsSettingsBulkActions,
+    handleSave,
+    setConversationsSettingsBulkActions,
+  ]);
+
+  const onDeleteCancelled = useCallback(() => {
+    setDeletedConversation(null);
+    closeConfirmModal();
+    onCancelClick();
+  }, [closeConfirmModal, onCancelClick]);
+
+  const { getConversationsList, getColumns } = useConversationsTable();
+
+  const conversationOptions = getConversationsList({
+    allSystemPrompts,
+    actionTypeRegistry,
+    connectors,
+    conversations: conversationSettings,
+    defaultConnector,
+  });
+
+  const onSaveCancelled = useCallback(() => {
+    closeEditFlyout();
+    onCancelClick();
+  }, [closeEditFlyout, onCancelClick]);
+
+  const onSaveConfirmed = useCallback(() => {
+    closeEditFlyout();
+    handleSave(true);
+    setConversationsSettingsBulkActions({});
+  }, [closeEditFlyout, handleSave, setConversationsSettingsBulkActions]);
+
+  const columns = useMemo(
+    () =>
+      getColumns({
+        conversations: conversationSettings,
+        onDeleteActionClicked,
+        onEditActionClicked,
+        allSystemPrompts,
+      }),
+    [allSystemPrompts, conversationSettings, getColumns, onDeleteActionClicked, onEditActionClicked]
+  );
+
+  const sorting = useMemo(
+    () => ({
+      sort: {
+        field: 'updatedAt',
+        direction: 'desc' as const,
+      },
+    }),
+    []
+  );
+
+  const confirmationTitle = useMemo(
+    () =>
+      deletedConversation?.title
+        ? i18n.DELETE_CONVERSATION_CONFIRMATION_TITLE(deletedConversation?.title)
+        : i18n.DELETE_CONVERSATION_CONFIRMATION_DEFAULT_TITLE,
+    [deletedConversation?.title]
+  );
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: DEFAULT_PAGE_INDEX,
+      pageSize: DEFAULT_PAGE_SIZE,
+      pageSizeOptions: [DEFAULT_PAGE_SIZE],
+      totalItemCount: conversationOptions.length,
+    }),
+    [conversationOptions.length]
+  );
```

Comment:
Looks like the pagination selector isn't working here for conversations. I tried with 10+ conversations (min page size) as well and am seeing same behavior:

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/915455ea-a734-4b25-871d-3a64ead3f2c8" />
</p>

Reply:
Yeah I found this bug as well, but after fixing it, I found another behaviour might look like a bug:

Currently we do not preserve the current pagination info for each table. The pagination info will be reset when switching between the tab.



https://github.com/elastic/kibana/assets/6295984/2e68c95e-c9a0-43e6-b15a-edd59a874976





Would it be better not to offer page options?



https://github.com/elastic/kibana/assets/6295984/9e8c28bf-7464-4d4c-b12b-ecfa093351cb

Reply:
> Currently we do not preserve the current pagination info for each table. The pagination info will be reset when switching between the tab.

I think this is fine for now. We could store in session storage though -- it's just a hook, so pretty easy to add via `import useSessionStorage from 'react-use/lib/useSessionStorage'`;

> Would it be better not to offer page options?

Nah, I think we should keep pagination 👍

Reply:
Anonymization default to 10 items per page, rest of the tables default to 25 items per page.
Pagination and sorting information are stored in session storage now:



https://github.com/elastic/kibana/assets/6295984/33af6e9e-11fe-4665-9eeb-a7767c4f3844